components/ai_chat/core/browser/code_sandbox_script_storage.h

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_AI_CHAT_CORE_BROWSER_CODE_SANDBOX_SCRIPT_STORAGE_H_
+#define BRAVE_COMPONENTS_AI_CHAT_CORE_BROWSER_CODE_SANDBOX_SCRIPT_STORAGE_H_
+
+#include <optional>
+#include <string>
+#include <string_view>
+
+#include "base/containers/flat_map.h"
+
+namespace ai_chat {
+
+// Class to manage script storage for code execution requests
+class CodeSandboxScriptStorage {
```

Comment:
why is this class necessary? It seems like a memory leak that could grow without bounds.

Reply:
I created this to avoid increasing the code size of the `AIChatService`. We could enforce a maximum size to the map if that would address any memory-related concerns.

Reply:
Removed the class and moved the single member/methods to `AIChatService`